robottelo/hosts.py

```diff
@@ -1815,6 +1817,19 @@ class DecClass(cls):
         self._api._configured = True
         return self._api
 
+    @property
+    def apidoc(self):
+        """Provide Satellite's apidoc via apypie"""
+        if not self._apidoc:
+            self._apidoc = apypie.Api(
+                uri=self.url,
+                username=settings.server.admin_username,
+                password=settings.server.admin_password,
+                api_version=2,
+                verify_ssl=False,
+            ).apidoc
+        return self._apidoc
```

Comment:
@JacobCallahan Do you see any benefit in making it a `@cached_property` instead of the current design?